### DISCUSSION TITLE
cgen documentation / examples

### DISCUSSION THREAD

**Discussion by max22-** [janet-lang/janet#1407] [Q&A] [answered]
Hi,
i am looking for some documentation or examples on how to generate C code using cgen.
I've found those 2 projects that use it : [best-self](https://git.sr.ht/~pepe/best-self/) and [shriek](https://git.sr.ht/~pepe/shriek/).
Are there any others ?

and a little question : what is the best way to generate this macro ?
`
#define ARRAY_SIZE(x) (sizeof(x)/sizeof((x)[0]))
`

Thanks,
Maxime

**Comment by pepe** [comment, marked as answer]
Hello. Author of both these libs here. I would suggest you too take a look at cjanet module of the spork library. 

And for example of its usage you may take a look at the code in https://git.sr.ht/~pepe/gp/tree/master/item/cjanet.

**Comment by max22-** [comment]
ok thank you very much !!

**Reply by pepe** [reply]
You are welcome. If you have more questions, please let me know. The best place is the Zulip as of now.
